### PR TITLE
Replace ModInv shim with KaliskiModInverse in ECAdd bloq

### PR DESCRIPTION
**Pull request by fpapa250**
Removes ModInv shim as it is no longer used
Replaces it with KaliskiModInverse bloq in ECAdd
Improves the gate count test to match toffoli count expected with bloq swap

### CODE REVIEW

**Comment by fpapa250** [comment]
https://github.com/quantumlib/Qualtran/issues/1443 - this issue is fixed now because we use KaliskiModInverse for classical simulation instead of my hacky temp solution.

File: qualtran/bloqs/factoring/ecc/ec_add.py [hunk 1]
```diff
@@ -285,7 +286,7 @@ def build_composite_bloq(
         ctrl, b, y = bb.add(CModSub(QMontgomeryUInt(self.n), mod=self.mod), ctrl=ctrl, x=b, y=y)
 
         # Perform modular inversion s.t. x = (x - a)^-1 % p.
-        x, z1, z2 = bb.add(ModInv(n=self.n, mod=self.mod), x=x)
+        x, z = bb.add(KaliskiModInverse(bitsize=self.n, mod=self.mod), x=x)
```
Comment (NoureldinYosri) [inline qualtran/bloqs/factoring/ecc/ec_add.py:289]: nit: junk

**Review by NoureldinYosri — APPROVED** [review]

File: qualtran/bloqs/factoring/ecc/ec_add_test.py [hunk 2]
```diff
@@ -407,19 +407,18 @@ def test_ec_add_symbolic_cost():
...
-    assert (
-        cost['n_ccz'].subs(m, n / 4).expand()
-        == sympy.Rational(253, 2) * n**2 + sympy.Rational(351, 2) * n - 35
-    )
+    # The following formula is 126.5n^2 + 195.5n - 31. We account for the discrepancy in the
+    # coefficient of n by a reduction in the toffoli cost of Montgomery ModMult, an increase in the
+    # toffoli cost for Kaliski Mod Inverse, n extra toffolis in ModNeg, 2n extra toffolis to do n
+    # 3-controlled toffolis in step 2. The expression is written with rationals because sympy
+    # comparison fails with floats.
+    assert total_toff == sympy.Rational(253, 2) * n**2 + sympy.Rational(391, 2) * n - 31
```
Comment (mpharrigan) [inline qualtran/bloqs/factoring/ecc/ec_add_test.py:421]: something that might help here or in the future: when comparing sympy expressions, the equality operator tests for the same literal expression. If you want to test for mathematical equality `x = y` you can test `simplify(x-y)==0`

**Review by mpharrigan — APPROVED** [review]
sweeeeeeeet

**Pull request merged**